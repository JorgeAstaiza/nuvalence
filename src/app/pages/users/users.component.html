<div class="container">
  <mat-form-field>
    <mat-label>Total users</mat-label>
    <mat-select
      (selectionChange)="totalUsersSelected($event)"
      [(value)]="totalListValue"
    >
      <mat-option *ngFor="let user of totalUsers" [value]="user.value">
        {{ user.viewValue }}
      </mat-option>
    </mat-select>
  </mat-form-field>
  <app-users-list
    [users]="userList"
    (userDetailsSelected)="userDetailsSelected($event)"
    (getUsers)="getUsers()"
  ></app-users-list>
</div>
