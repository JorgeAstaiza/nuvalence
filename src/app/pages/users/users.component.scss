.container {
    width: 70%;
    margin: 0 auto;
    display: flex;
    flex-direction: column;
    flex-wrap: wrap;
    justify-content: center;
    height: 100vh;
}