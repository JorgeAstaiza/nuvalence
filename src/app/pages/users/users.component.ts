import { Component, EventEmitter, Output } from '@angular/core';
import { MatSelectChange } from '@angular/material/select';
import { Router } from '@angular/router';
import { Result } from 'src/app/shared/interface/user.interface';
import { UsersService } from '../../../../src/app/shared/services/users.service';

@Component({
  selector: 'app-users',
  templateUrl: './users.component.html',
  styleUrls: ['./users.component.scss'],
})
export class UsersComponent {
  @Output() totalUser = new EventEmitter<number>();
  totalListValue = 10;
  totalUsers = [
    { value: 10, viewValue: 'list of 10 users' },
    { value: 20, viewValue: 'list of 20 users' },
    { value: 30, viewValue: 'list of 30 users' },
    { value: 40, viewValue: 'list of 40 users' },
    { value: 50, viewValue: 'list of 50 users' },
  ];

  userList!: Result[];

  constructor(private userService: UsersService, private router: Router) {}

  public totalUsersSelected(e: MatSelectChange) {
    this.totalListValue = e.value;
    this.getUsers();
  }

  public userDetailsSelected(user: Result) {
    console.log(user);
    this.userService.updateUserDetails(user);
    this.router.navigate(['/details']);
  }

  public getUsers() {
    this.userService
      .getUsers(this.totalListValue, 'nuvalence')
      .subscribe((res) => {
        console.log(res.results);
        this.userList = res.results;
      });
  }
}
