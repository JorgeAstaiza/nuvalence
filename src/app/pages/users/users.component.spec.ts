import { ComponentFixture, TestBed } from '@angular/core/testing';
import {
  HttpClientTestingModule,
  HttpTestingController,
} from '@angular/common/http/testing';

import { UsersComponent } from './users.component';
import { UsersService } from '../../../../src/app/shared/services/users.service';
import { mockUsers } from '../../../../src/app/shared/services/mock.users';
import { MatSelectChange } from '@angular/material/select';
import { of } from 'rxjs';

describe('UsersComponent', () => {
  let component: UsersComponent;
  let fixture: ComponentFixture<UsersComponent>;
  let compiled: HTMLElement;
  let userService: UsersService;
  let httpMock: HttpTestingController;

  beforeEach(() => {
    TestBed.configureTestingModule({
      declarations: [UsersComponent],
      imports: [HttpClientTestingModule],
      providers: [UsersService],
    });
    fixture = TestBed.createComponent(UsersComponent);
    component = fixture.componentInstance;
    userService = TestBed.inject(UsersService);
    httpMock = TestBed.inject(HttpTestingController);
    fixture.detectChanges();
    compiled = fixture.nativeElement;
  });

  test('should create', () => {
    expect(component).toBeTruthy();
  });

  test('should do match with snapshot', () => {
    expect(compiled.innerHTML).toMatchSnapshot();
  });

  test('should get users', () => {
    userService.getUsers(5, 'nuvalence').subscribe(() => {
      const req = httpMock.expectOne(
        'https://randomuser.me/api?results=5&seed=nuvalence'
      );
      expect(req.request.method).toBe('GET');

      // Completa la solicitud simulando una respuesta exitosa
      req.flush(mockUsers);

      // Verifica que userList se haya actualizado correctamente
      expect(component.userList).toEqual(mockUsers.results);
    });
  });

  it('should handle totalUsersSelected event', () => {
    const getUsersSpy = jest.spyOn(component, 'getUsers');
    // Simula el evento de selecci√≥n de usuarios
    const event = { value: '10' } as MatSelectChange;
    component.totalUsersSelected(event);
    expect(component.totalListValue).toBe('10');
    expect(getUsersSpy).toHaveBeenCalled();
  });
});
