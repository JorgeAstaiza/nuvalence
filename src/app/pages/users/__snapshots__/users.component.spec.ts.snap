// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsersComponent should do match with snapshot 1`] = `
"<div class="container"><mat-form-field><mat-label>Total users</mat-label><mat-select><mat-option> list of 10 users </mat-option><mat-option> list of 20 users </mat-option><mat-option> list of 30 users </mat-option><mat-option> list of 40 users </mat-option><mat-option> list of 50 users </mat-option><!--bindings={
  "ng-reflect-ng-for-of": "[object Object],[object Object"
}--></mat-select></mat-form-field><app-users-list></app-users-list></div>"
`;
